sdk/cosmos/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/repository/ReactiveCourseRepository.java

```diff
@@ -34,4 +35,12 @@
      * @return Course list
      */
     Flux<Course> findByNameOrDepartmentAllIgnoreCase(String name, String department);
+
+    /**
+     * Find a single Course list by name
+     * @param name name
+     * @return Course list
+     */
+    Mono<Course> findByName(String name);
```

Comment:
Shouldn't this also be `findOneByName` ?
I am still unclear on what if there are multiple items by `name` ?

Reply:
yea i will rename that for clarity. But i do want to point out that the determination of the execution path is based on the return type. We are not using the PartTree class to parse the method name to determine the users intent so it behaves the same regardless of whether the user specifies "findOne". The fact that the method is returning a Mono is what makes it expect one item.

In the event that there are multiple items returned, we will throw a CosmosAccessException with an error message saying "Too many results - Expected Mono<Course> but query returned multiple results".

This is consistent with the non reactive path and is demonstrated in the test ReactiveCourseRepositoryIT.testFindOneShouldFailIfMultipleResultsReturned

Reply:
Sounds great, thank you for the explanation.